pkg/deploy/controller/deployment/factory.go

```diff
@@ -110,6 +111,10 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 					return false
 				}
 				if retries.Count > 1 {
+					deployment, ok := obj.(*kapi.ReplicationController)
+					if ok {
+						eventRecorder.Eventf(deployment, kapi.EventTypeWarning, "FailedRetry", "About to stop retrying %s: %v", deployment.Name, err)
```

Comment:
I am not sure stable events on transient errors is possible. I agree that most of the retriable events aren't meaningful but note that those will be emitted on the underlying rc and not the dc. Anyway, I vetted all the errors returned and the only actionable error worth emitting an event for is when the deployer pods won't be cleaned up. Updated, PTAL